newIDE/app/src/BehaviorsEditor/Editors/Physics2Editor/index.js

```diff
@@ -359,43 +367,45 @@ const Physics2Editor = (props: Props) => {
               project={props.project}
               resourcesLoader={resourcesLoader}
               fixedHeight={200}
-              renderOverlay={overlayProps => (
-                <ShapePreview
-                  shape={properties.get('shape').getValue()}
-                  dimensionA={parseFloat(
-                    properties.get('shapeDimensionA').getValue()
-                  )}
-                  dimensionB={parseFloat(
-                    properties.get('shapeDimensionB').getValue()
-                  )}
-                  offsetX={parseFloat(
-                    properties.get('shapeOffsetX').getValue()
-                  )}
-                  offsetY={parseFloat(
-                    properties.get('shapeOffsetY').getValue()
-                  )}
-                  polygonOrigin={properties.get('polygonOrigin').getValue()}
-                  vertices={JSON.parse(properties.get('vertices').getValue())}
-                  width={overlayProps.imageWidth}
-                  height={overlayProps.imageHeight}
-                  frameOffsetTop={overlayProps.offsetTop}
-                  frameOffsetLeft={overlayProps.offsetLeft}
-                  zoomFactor={overlayProps.imageZoomFactor}
-                  onMoveVertex={(index, newX, newY) => {
-                    let vertices = JSON.parse(
-                      properties.get('vertices').getValue()
-                    );
-                    vertices[index].x = newX;
-                    vertices[index].y = newY;
-                    behavior.updateProperty(
-                      behaviorContent.getContent(),
-                      'vertices',
-                      JSON.stringify(vertices)
-                    );
-                    forceUpdate();
-                  }}
-                />
-              )}
+              renderOverlay={overlayProps => {
+                return (

```

Comment:
Try to `const properties = behavior.getProperties(behaviorContent.getContent());` and add a comment telling that `the result from getProperties is a temporary, and because this renderOverlay function can be called outside of the render, we must get the properties again.`